scripts/update-requirements

```diff
@@ -31,20 +38,43 @@ def main():
     if args.project is not None:
         project_name = args.project
     else:
-        project_name = utils.project_name(pkg_dir)
+        project_name = utils.get_project_name(pkg_dir)
     requirements_file = pkg_dir / args.requirements / "requirements.txt"
+    if not requirements_file.exists():
+        poetry_lock = requirements_file.parent.parent / "poetry.lock"
+        pyproject_toml = requirements_file.parent.parent / "pyproject.toml"
+        if poetry_lock.exists() and pyproject_toml.exists():
+            requirements_file = poetry_lock
+        else:
+            print("Could not find project requirements, checked:")
+            print(f"* {requirements_file.absolute()}")
+            print(f"* {poetry_lock.absolute()}")
+            sys.exit(1)
+
     project = Path(__file__).parent.parent / project_name
 
     # First remove index line and any PyQt or sip dependency
-    cleaned_lines = cleanup(requirements_file)
+
+    if requirements_file.name == "poetry.lock":
+        dependencies = utils.get_poetry_names_and_versions(
+            requirements_file, pyproject_toml
+        )
+    else:
+        dependencies = utils.get_requirements_names_and_versions(requirements_file)
+
+    dependencies = [dep for dep in dependencies if dep[0] not in SKIP_PACKAGES]
 
     verify_sha256sums_file(project)
 
+    # If there is no 'requirements' directory, we look in the project root
+    if not (pkg_dir / args.requirements).exists():
+        args.requirements = ""
```

Comment:
This feels a bit fragile and it would be nice to avoid mutating `args`.

Maybe something like:
```python
if (pkg_dir / args.requirements).exists():
    build_requirements_file = pkg_dir / args.requirements / "build-requirements.txt"
else:
    # If there is no 'requirements' directory, we look in the project root
    build_requirements_file = pkg_dir / "build-requirements.txt"
```